notebooks/Notebook3.md

```diff
@@ -2,33 +2,22 @@
 
-**TODO:** Fill in this part with a summary and reflection on the critique you received for
-last week's work. Answer questions such as:  How, specifically, did the feedback help
-improve the project? Did the feedback point out or offer something you hadn't considered?
-Did it help you make a design decision? Was it helpful in addressing the most pressing
-issues in your project? How will you incorporate the feedback into your work? Will you
-change something about the design, implementation, or evaluation as a result?
+<p>My feedback partner recommended that I borrow my design from css, so I took that advice at least for creating shapes in my language because I thought that it would be good for describing the attributes of an element. Also, the process of talking to my feedback group about how they hade written their parsers helped me greatly with planning out writing my parser. Some of the questions that my feedback partner posed when he was discussing struggles with his own language regarding outputting javascript files from his parser helped me do some research on that before I started working on my parser.</p>
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+<p>I started working on [my parser](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/DSL%20Code/parser.py) this week, so I also made adjustments to my language design to make it so that my parser could get to my desired result better. So far making the parser doesn't seem too hard. I was able to output an extendscript file that selected a composition to work on. Currently I'm able to process code that selected the current composition that is selected in after effects or that pics a specific composition with a given index. </p>
+
+<p>This is the [output I've gotten so far](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/select_comp.png) from my parser, and this is my [goal output](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/goal_output.png). I'm going to make sure that my parser can catch errors, I already added an error catch that tells the user to make sure that their code picks a composition in the first line of code.</p>
+
+<p>I am nervous that the way that I've been writing my parser is incorrect because it seems to simple or that the way I've written my parser is too prone to mistakes. If I find out that I've done it the wrong way, it's going to push back my progress. I'm treating the file that that I write my DSL in as a txt file, and that text file is just read into the parser line by line. When it reads those lines in, it just translates it to extenscript.</p>
```

Comment:
Instead of reading line-by-line, you can also read the entire file into a string using `file_string = f.read()` after your `f = open(file, "r")` line. That way you can easily split it into an array by line (`file_string.split("\n")`), or maybe more usefully, into an organized structure using regex.

Reply:
You can also do `f.readlines()` to get a list of the lines.